snuba/datasets/events_format.py

```diff
@@ -100,13 +100,25 @@ def extract_extra_contexts(
     return (context_keys, context_values)
 
 
-def enforce_retention(message: Mapping[str, Any], timestamp: Optional[datetime]) -> int:
+def override_and_enforce_retention(
+    message: Mapping[str, Any], timestamp: Optional[datetime]
```

Comment:
Nit: I would prefer if you passed in the parameter `project_id` and `retention_days` over passing the full message. Those fields could be called something else (like `projectid`) in some other dataset in the future. And let's assume we want to override and enforce this in the new dataset. Passing the message would fail in those cases.